src/components/com_tjcertificate/site/views/certificate/view.html.php

```diff
@@ -76,23 +79,24 @@ public function display($tpl = null)
 
 		if (!$this->certificate)
 		{
-			$app = Factory::getApplication();
 			$app->enqueueMessage(Text::_('COM_TJCERTIFICATE_ERROR_CERTIFICATE_EXPIRED'), 'error');
+
+			$app->redirect($redirectBackUrl);
```

Comment:
Is it a good idea to handle redirects in the view file @ankush-maherwal ?

Reply:
@coolbung We can use redirects in view.html.php i.e view files but we should not write the business logic in the view file which determines whether to redirect the user or allow the user to proceed further. Eg If s user is not authorized to view any page then we can redirect him from the view file but the business logic which determines the authorization should be in the respective model file